plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/validate-state.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { ShippingAddress } from '@woocommerce/settings';
+import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
+import { __ } from '@wordpress/i18n';
+
+function previousString( initialValue?: string ) {
```

Comment:
Since there are no hooks in the other validation I wanted to keep the same approach here. This simple function keeps track of the previous value because we need to know if the country changed when determining to show validation of the state.

Reply:
@opr yeah I did want to do that refactoring too tbh. Maybe it's worth doing here and now.